packages/node_modules/@ciscospark/react-component-adaptive-card/src/index.js

```diff
@@ -22,6 +26,36 @@ class AdaptiveCard extends React.Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
+    if (prevProps !== this.props && this.props.verb === API_ACTIVITY_VERB.SHARE) {
+      try {
+        this.cards[0] = getDecryptedURls(this.props);
+        if (this.cards[0]) {
+          this.setupState();
+        }
+      }
+      catch (error) {
+        this.props.sparkInstance.logger.error('Unable render Adaptive Card', error.message);
+      }
+    }
+
+    return null;
+  }
+
+  componentWillUnmount() {
+    this.state.childNodes.forEach((childNode) => {
+      ReactDOM.unmountComponentAtNode(childNode);
+    });
+  }
+
+  /**
+   * set state to tohasReplacedImagesInJSON
+   * @returns {void}
+   */
+  setupState() {
```

Comment:
Is this called more than in one place? does it need to be a method?

Reply:
we can not set the state in `componentDidUpdate` lifecycle component so that's the reason for setting the state in method